src/Update/Nix/FetchGit/Utils.hs

```diff
@@ -72,6 +74,14 @@ deltaToSourcePos :: Delta -> SourcePos
 deltaToSourcePos delta = SourcePos line column
                  where (Directed _ line column _ _) = delta
 
+-- | Given an expression that is supposed to represent a function,
+-- extracts the name of the function.  If we cannot figure out the
+-- function name, returns Nothing.
+extractFuncName :: NExprLoc -> Maybe Text
+extractFuncName (AnnE _ (NSym name)) = Just name
+extractFuncName (AnnE _ (NSelect _ (reverse -> StaticKey name:_) _)) = Just name
```

Comment:
Could use `lastMay` here instead of reverse.

https://hackage.haskell.org/package/safe-0.3.9/docs/Safe.html#v:lastMay